### ISSUE TITLE
How to get client ip?

### ISSUE THREAD

**Issue by xfxpositions** [elysiajs/elysia#230]
i'm trying to find something like this
from koajs
 ```ts 
const client_ip =
ctx.req.ip ||
ctx.req.headers["x-forwarded-for"] ||
ctx.req.connection.remoteAddress;
```

**Comment by masfahru** [comment]
i think client ip should be available after this PR https://github.com/oven-sh/bun/pull/6165 is done

**Comment by bogeychan** [comment]
Hi 👋

since [Bun v1.0.4](https://bun.sh/blog/bun-v1.0.4#implement-server-requestip) you can do something like this:

```ts
import { Elysia } from 'elysia';

const app = new Elysia();

app
  .derive((ctx) => ({
    remoteAddress: () => app.server!.requestIP(ctx.request)
  }))
  .get('/', (ctx) => ctx.remoteAddress())

  .listen(3000);

console.log(`Listening on http://${app.server!.hostname}:${app.server!.port}`);
```

Response:
```json
{
    "address": "::1",
    "family": "IPv6",
    "port": 40908
}
```

**Comment by juancwu** [comment]
Is there a plan to add the IP to the context request object by default? If not, I am down to try to add it. I am relatively new here but I am sure it shouldn't be too hard.

Please let me know if having it in the context request object is desired or not :)

**Comment by gaurishhs** [comment]
You can use https://github.com/gaurishhs/elysia-ip

**Comment by cassinius** [comment]
Thank you for providing this library, it works well on the main Elysia app instance ;-)

However, if I try to use this in a plugin, I get

`null is not an object (evaluating '<plugin>.server.requestIP')`

Is there any way to make this work in a specific plugin only ?

**Comment by gaurishhs** [comment]
that means the server isn't up yet, Try to use the plugin inside an Elysia app instance and call the .listen method, It should work well. Currently, You don't seem to call .listen

**Comment by cassinius** [comment]
I am calling .listen on the main Elysia app instance, here is my code:

```
import { Elysia } from "elysia";
import { ip } from "elysia-ip";

const plugin = new Elysia()
  .use(ip())
  .get("/ip-from-plugin", ({ ip }) => ip);

const app = new Elysia()
  .use(ip())
  .get("/ip", ({ ip }) => ip)
  .use(plugin)
  .listen(3000);
```
Here, `/ip` works as expected, `/ip-from-plugin` throws the error.
If I call .listen(<different_port>) on the plugin as well, like
```
const plugin = new Elysia()
  .use(ip())
  .get("/ip-from-plugin", ({ ip }) => ip)
  .listen(3001);
```
it actually DOES work, but I doubt that's how it's meant to be used?

P.S. removing the ip-related lines from the main instance does not make a difference.

**Comment by gaurishhs** [comment]
I am facing the same, i don't really know the reason behind the same.

**Comment by gaurishhs** [comment]
@cassinius Hi, i seem to have found a fix for the same. Not initalizing the ip in plugin seems to work. 

```ts
import { Elysia } from "elysia";
import { ip } from "elysia-ip";

const plugin = new Elysia()
  // .use(ip())
  .get("/ip-from-plugin", ({ ip }) => ip);

const app = new Elysia()
  .use(ip())
  .get("/ip", ({ ip }) => ip)
  .use(plugin)
  .listen(3000);
  ```
  
  The autocomplete is not there for some reason in plugin and there on the app instance which doesn't really make sense to me. I reached out to salty for further help.

**Comment by xfxpositions** [comment]
```ts
// getting request ip
app.use(ip());
```
```sh
TypeError: undefined is not an object (evaluating 'request.headers')
```
not working

**Comment by gaurishhs** [comment]
Can you share your complete code and how youre making the request. The
headers shouldnt be undefined though.

**Comment by SaltyAom** [comment]
Close as [elysia-ip](https://github.com/gaurishhs/elysia-ip) plugin and `server` property is now implemented in Elysia 1.1

You can natively get IP from Elysia request like this, make sure server is running:
```ts
new Elysia()
	.get('/', ({ server, request }) => {
		server?.requestIP(request)
	})
        .listen(3000)
```
